WaterLog_Backend/Controllers/SegmentLeaksController.cs

```diff
@@ -0,0 +1,134 @@
+        public async Task<ActionResult<string>> GetCost(int id)
+        {
+            SegmentLeaksEntry leaks = _db.SegmentLeaks.Find(id);
+            if (leaks == null)
+            {
+                return NotFound();
+            }
+            Procedures procedures = new Procedures(_service);
+            return (JsonConvert.SerializeObject((procedures.calculateTotalCost(leaks), procedures.calculatePerHourCost(leaks))));
+        }
+
+        [Route("litres/{id}")]
+        public async Task<ActionResult<string>> GetLitres(int id)
+        {
+            SegmentLeaksEntry leaks = _db.SegmentLeaks.Find(id);
+            if(leaks == null)
+            {
+                return NotFound();
+            }
+            Procedures procedures = new Procedures(_service);
+       
+            return (JsonConvert.SerializeObject((procedures.calculateTotaLitres(leaks), procedures.calculateLitresPerHour(leaks))));
+        }
+
+        //Resolve Leakage
+        [HttpPost("resolve")]
+        public async Task<ActionResult<SegmentLeaksEntry>> Resolve([FromForm] int id)
+        {
+            var leaks = await _db.SegmentLeaks.FindAsync(id);
+            if(leaks == null)
+            {
+                return NotFound();
+            }
+            leaks.ResolvedStatus = "resolved";
+            await _db.SaveChangesAsync();
+            return leaks;
+
+        }
+        // GET api/values/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<SegmentLeaksEntry>> Get(int id)
+        {
+            var leaks = await _db.SegmentLeaks.FindAsync(id);
+            
+            if(leaks == null)
+            {
+                return NotFound();
+            }
+
+            return leaks;
+        }
+       
+        // POST api/values
+        [HttpPost]
+        public async Task Post([FromBody] SegmentLeaksEntry value)
+        {
+            await _db.SegmentLeaks.AddAsync(value);
+            await _db.SaveChangesAsync();
+
+            ;
+        }
+
+        // PUT api/values/5
+        [HttpPut("{id}")]
+        public async Task Put(int id, [FromBody] SegmentLeaksEntry value)
+        {
+            var entry = await _db.SegmentLeaks.FindAsync(id);
+            entry = value;
+            await _db.SaveChangesAsync();
+        }
+
+        // DELETE api/values/5
+        [HttpDelete("{id}")]
+        public async Task Delete(int id)
+        {
+            var entry = await _db.SegmentLeaks.FindAsync(id);
+            _db.SegmentLeaks.Remove(entry);
+            await _db.SaveChangesAsync();
+        }
+
+        [HttpPatch("{id}")]
+        public async Task Patch([FromBody] SegmentLeaksEntry value)
+        {
+            var entry = _db.SegmentLeaks.FirstOrDefault(segL => segL.Id == value.Id);
+
+            if(entry != null)
+            {
+                entry.LatestTimeStamp = value.LatestTimeStamp;
+               await _db.SaveChangesAsync();
+            }
+            
+        }
+    }
+}
```

Comment:
Please ensure empty line at the end of files

Reply:
done